base-permissions/src/commonMain/kotlin/PermissionStateRepo.kt

```diff
@@ -88,14 +91,16 @@ open class PermissionStateRepo<P : Permission>(
 
     private val superVisorJob = SupervisorJob(coroutineContext[Job])
     private suspend fun startMonitoringManager(permissionManager: PermissionManager<P>) {
+        val hasStarted = EmptyCompletableDeferred()
         CoroutineScope(coroutineContext + superVisorJob).launch {
-            permissionManager.events.collect { event ->
+            permissionManager.events.onStart { hasStarted.complete() }.collect { event ->
```

Comment:
What does this fix? (I removed it and it seemed to have no effect).

I see it waits until collection start, but since this is a Channel won't the events just be buffered if the collect does not immediately run?